bundles/org.openhab.core.config.discovery.addon/src/main/java/org/openhab/core/config/discovery/addon/finders/MDNSAddonSuggestionFinder.java

```diff
@@ -0,0 +1,182 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.config.discovery.addon.finders;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.jmdns.ServiceEvent;
+import javax.jmdns.ServiceInfo;
+import javax.jmdns.ServiceListener;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.addon.AddonDiscoveryMethod;
+import org.openhab.core.addon.AddonInfo;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.io.transport.mdns.MDNSClient;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a {@link MDNSAddonSuggestionFinder} for finding suggested Addons via MDNS.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = AddonSuggestionFinder.class, name = MDNSAddonSuggestionFinder.SERVICE_NAME, configurationPid = MDNSAddonSuggestionFinder.CONFIG_PID)
+public class MDNSAddonSuggestionFinder extends BaseAddonSuggestionFinder implements ServiceListener {
+
+    public static final String SERVICE_TYPE = "mdns";
+    public static final String SERVICE_NAME = SERVICE_TYPE + ADDON_SUGGESTION_FINDER;
+    public static final String CONFIG_PID = ADDON_SUGGESTION_FINDER_CONFIG_PID + SERVICE_TYPE;
+
+    private static final String NAME = "name";
+    private static final String APPLICATION = "application";
+
+    private final Logger logger = LoggerFactory.getLogger(MDNSAddonSuggestionFinder.class);
+    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(SERVICE_NAME);
+    private final Map<String, ServiceInfo> services = new ConcurrentHashMap<>();
+    private final MDNSClient mdnsClient;
+
+    @Activate
+    public MDNSAddonSuggestionFinder(@Nullable Map<String, Object> configProperties, @Reference MDNSClient mdnsClient) {
+        this.mdnsClient = mdnsClient;
+        activate(configProperties);
+    }
+
+    /**
+     * Adds the given mDNS service to the set of discovered services.
+     * 
+     * @param device the mDNS service to be added.
+     */
+    public void addService(ServiceInfo service, boolean isResolved) {
+        String qualifiedName = service.getQualifiedName();
+        if (isResolved || !services.containsKey(qualifiedName)) {
+            if (services.put(qualifiedName, service) == null) {
+                logger.trace("Added service: {}/{}", qualifiedName, service.getNiceTextString());
```

Comment:
Looking at my trace log, not sure the second part of the log adds much.
```
16:53:36.078 [TRACE] [on.finders.MDNSAddonSuggestionFinder] - Added service: FP00112A22751D._http._tcp.local./ FP00112A22751D._http._tcp.local.
16:53:36.080 [TRACE] [on.finders.MDNSAddonSuggestionFinder] - Added service: 04255CB411F0@Volumio._raop._tcp.local./&04255CB411F0@Volumio._raop._tcp.local.
16:53:36.080 [TRACE] [on.finders.MDNSAddonSuggestionFinder] - Added service: volumio._Volumio._tcp.local./\034volumio._Volumio._tcp.local.
16:53:36.125 [TRACE] [on.finders.MDNSAddonSuggestionFinder] - Added service: SCC-00000874._http._tcp.local./\036SCC-00000874._http._tcp.local.
16:53:36.132 [TRACE] [on.finders.MDNSAddonSuggestionFinder] - Added service: HP Color LaserJet MFP M281fdw (0CACA4)._http._tcp.local./8HP Color LaserJet MFP M281fdw (0CACA4)._http._tcp.local.
16:53:36.132 [TRACE] [on.finders.MDNSAddonSuggestionFinder] - Added service: HP Color LaserJet MFP M281fdw (0CACA4)._ipp._tcp.local./7HP Color LaserJet MFP M281fdw (0CACA4)._ipp._tcp.local.
16:53:36.132 [TRACE] [on.finders.MDNSAddonSuggestionFinder] - Added service: HP Color LaserJet MFP M281fdw (0CACA4)._printer._tcp.local./;HP Color LaserJet MFP M281fdw (0CACA4)._printer._tcp.local.
16:53:36.726 [TRACE] [on.finders.MDNSAddonSuggestionFinder] - Added service: NEEO-d487672e._hap._tcp.local./\036NEEO-d487672e._hap._tcp.local.
16:53:36.768 [TRACE] [on.finders.MDNSAddonSuggestionFinder] - Added service: NEEO Zithoek._neeo._tcp.local./\036NEEO Zithoek._neeo._tcp.local.
```

Reply:
Fixed.